lnrpc/routerrpc/router_server.go

```diff
@@ -1531,6 +1544,138 @@ func (s *Server) HtlcInterceptor(stream Router_HtlcInterceptorServer) error {
 	).run()
 }
 
+// XAddLocalChanAliases is an experimental API that creates a set of new
+// channel SCID alias mappings. The list of successfully created mappings is
+// returned. This is only a locally stored alias, and will not be communicated
+// to the channel peer via any message. Therefore, routing over such an alias
+// will only work if the peer also calls this same RPC on their end. If an
+// alias already exists, an error is returned.
+func (s *Server) XAddLocalChanAliases(_ context.Context,
+	in *AddAliasesRequest) (*AddAliasesResponse, error) {
+
+	existingAliases := s.cfg.AliasMgr.ListAliases()
+
+	// aliasExists checks if the new alias already exists in the alias map.
+	aliasExists := func(newAlias uint64,
+		baseScid lnwire.ShortChannelID) (bool, error) {
+
+		// First check that we actually have a channel for the given
+		// base scid. This should succeed for any channel where the
+		// option-scid-alias feature bit was negotiated.
+		if _, ok := existingAliases[baseScid]; !ok {
+			return false, fmt.Errorf("base scid %v not found",
+				baseScid)
+		}
+
+		for base, aliases := range existingAliases {
+			for _, alias := range aliases {
+				exists := alias.ToUint64() == newAlias
+
+				// Trying to add an alias that we already have
+				// for another channel is wrong.
+				if exists && base != baseScid {
+					return true, fmt.Errorf("%w: alias %v "+
+						"already exists for base scid "+
+						"%v", ErrAliasAlreadyExists,
+						alias, base)
+				}
+
+				if exists {
+					return true, nil
+				}
+			}
+		}
+
+		return false, nil
+	}
```

Comment:
IMO not in this case. Since this allows us to operate on the cached list of existing aliases. With a method on the alias manager we'd query more often, holding the lock more often.